File: src/12.ts
// Computes the value of the first triangle number to have over the given number of divisors.
//
// See {@link https://projecteuler.net/problem=12}.
import tau from './core/tau';

export default function compute(limit: number): number {
  let n = 0;
  let i = 1;

  while (tau(n) <= limit) {
    n += i;
    i += 1;
  }

  return n;
}

File: project_euler/12.py
# Problem 12: Highly Divisible Triangular Number
#
# What is the value of the first triangle number to have over five hundred divisors?
#
# See https://projecteuler.net/problem=12
from project_euler.common.divisors import tau


def run(limit=500) -> int:
    n = 0
    i = 1

    while tau(n) <= limit:
        n += i
        i += 1

    return n


def test_run():
    assert run() == 76576500
